assorted test-changes in preparation of updating integration-cli CLI version

thaJeztah opened the pull request:
- relates to https://github.com/moby/moby/pull/51361


**- Human readable description for the release notes**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section.

NOTE: Only fill this section if changes introduced in this PR are user-facing.
The PR must have a relevant impact/ label.
-->
```markdown changelog

```

**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
integration-cli: rename var that shadowed package-level const

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: DockerCLIBuildSuite: use more unique names for images

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: improve TestBuildNoNamedVolume

Make sure the error is the one we expected, and slightly clarify
the behavior (named volumes not having a meaning in VOLUME, so will
be used as-is).

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: rewrite some tests depending on intermediate images

These tests used intermediate images (as produced by the classic builder)
as part of the test. When using BuildKit, such images are not produced,
and will only be in the build-cache.

From the tests, it looks like the extra checks were not critical to
verify the behavior, so let's simplify them to not depend on this.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: use event-consts in some tests

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: adjust some tests for API-specific event-types

Image create events were added in API 1.46, and filtered out in older
API versions.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: TestUserDefinedNetworkAlias: adjust for API versions

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
integration-cli: TestUserDefinedNetworkAlias: replace deprecated utils

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

robmry commented on integration-cli/docker_cli_events_test.go:
```diff
@@ -588,7 +588,7 @@ func (s *DockerCLIEventSuite) TestEventsFilterType(c *testing.T) {
 		"--until", daemonUnixTime(c),
 		"--filter", fmt.Sprintf("label=%s", label),
-		"--filter", "type=build",
+		"--filter", "type="+string(eventtypes.BuilderEventType),
```
`eventtypes.BuilderEventType` is `"builder"` rather than `"build"` ... but the test passes with either. Hmm, is that a sign of, something?

thaJeztah replied:
Pushed a commit to remove that part of the test, as it had no value with Buildkit (which doesn't produce events during build).

robmry reviewed (approved):
LGTM - spotted an oddity, but not directly related to the change ...

thaJeztah committed:
integration-cli: TestEventsFilterType: remove test for "build" events

Commit 6aea8c2591d40c96d90a2da0a1d499799cdcabad updated this part of the
test to account for BuildKit now being used as default, which doesn't use
regular containers to "commit" to an image, and therefore doesn't produce
events as part of the build.

However, it changed it to filter for `build` type events, which .. don't
exist. The test passed because it only checked for number of events produced,
which is approximated by checking the number of lines in the output. In
this case, the output would be empty, so "events" was a slice of strings
with length 1; an empty string.

Given that BuildKit doesn't produce meaningful events to test for, we could
either disable BuildKit, to test the legacy builder, or just remove this
part of the test. Let's do the latter.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
